client/state/login/reducer.js

```diff
@@ -82,6 +83,8 @@ export const redirectTo = combineReducers( {
 				const { path, query } = action;
 				if ( startsWith( path, '/log-in' ) ) {
 					return query.redirect_to || state;
+				} else if ( '/jetpack/connect/authorize' === path ) {
+					return addQueryArgs( query, path );
```

Comment:
This is so the user gets dropped in the same URL after logging in, instead of defaulting to `/`.